conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEntryPoint.kt

```diff
@@ -5,14 +5,46 @@ import com.r3.conclave.common.internal.CallHandler
 import com.r3.conclave.common.internal.EnclaveCallType
 import com.r3.conclave.enclave.Enclave
 import com.r3.conclave.utilities.internal.getRemainingString
+import java.io.File
 import java.lang.NumberFormatException
 import java.nio.ByteBuffer
+import java.util.*
+import kotlin.properties.Delegates
 import kotlin.system.exitProcess
 
 object GramineEntryPoint {
     private const val USAGE_STRING = "usage: GramineEntryPoint <port>"
+    private const val ENCLAVE_METADATA_FILE = "enclave-metadata.properties"
     private const val EXIT_ERR = -1
 
+    /** Enclave metadata variables */
+    private var isSimulation by Delegates.notNull<Boolean>()
```

Comment:
What's Delegates?

Reply:
Alternative to `lateinit` that works for primitive types. Was actually suggested to me by the IDE!

Reply:
I agree, it's a bit odd. I would not trust IntelliJ sometimes.

Reply:
Removed, just used nullable values instead for both of them